src/frontend/Driver.ml

```diff
@@ -28,20 +28,38 @@ let elaborate_typed_term name (args : CS.cell list) tp tm =
   let+ vtm = RM.lift_ev @@ Sem.eval tm in
   tp, vtp, tm, vtm
 
-let execute_decl : CS.decl -> [`Continue | `Quit] RM.m =
+type status = (unit, unit) Result.t
+type continuation = [`Continue of status RM.m -> status RM.m | `Quit]
+
+let add_global name vtp con =
+  let open Monad.Notation (RM) in
+  let+ _ = RM.add_global name vtp con in
+  let kont = 
+    match vtp with
+    | D.TpPrf phi ->
+      RM.restrict [phi]
+    | _ -> 
+      (fun m -> m)
```

Comment:
There's `Fun.id` LOL not sure if it's better, though

Reply:
thanks!